docs/TELEMETRY.md

```diff
@@ -0,0 +1,119 @@
+# Telemetry system
+
+In Web Chat 4.8, a new telemetry system is introduced. Developers can opt into this feature by implementing their own telemetry collection adapter and start pumping measurements into telemetry services of their choice.
+
+Microsoft **does not** collect or receive any telemetry measurements for Web Chat on either CDN and NPM releases. Developers must provide their own telemetry collection adapter to start collecting data. Under local regulations, they may be required to provide privacy policies to the end-user to explain their data collection policy and provide a prompt.
+
+## Collection
+
+We have 2 samples for collecting telemetry measurements:
+
+-  [Collecting telemetry measurements using Azure Application Insights](https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/04.api/k.telemetry-application-insights)
+-  [Collecting telemetry measurements using Google Analytics](https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/04.api/l.telemetry-google-analytics)
+
+## Measurements
+
+We simplified our measurements to make it suitable for popular telemetry services:
+
+-  Dimension: dimensions are environment information, including browser capabilities and options passed to Web Chat
+-  Event: event will include a name, optional string data, and optional non-negative finite integer value
+-  Exception: exception will include the JavaScript error object
+-  Timing: timing will fire two events, `timingstart` and `timingend`
+   -  `timingstart` will include a name
+   -  `timingend` will include a name, and duration measured in milliseconds
+
+Feature updates on minor versions may introduce new or remove outdated measurements. To better understand user behaviors with different set of measurements, developers are advised to tag their telemetry data with the Web Chat version.
+
+### Dimensions
+
+The following information will be emitted on every measurement. Dimensions may change during the session.
+
+| Name                                   | Description                                                                              |
+| -------------------------------------- | ---------------------------------------------------------------------------------------- |
+| `prop:locale`                          | Locale specified in props, normalized                                                    |
+| `prop:speechRecognition`               | `"false"` if speech recognition is switched off                                          |
+| `prop:speechSynthesis`                 | `"false"` if speech synthesis is switched off                                            |
+| `capability:downscaleImage:workerType` | `"web worker"` if the browser support Web Worker and offline canvas, otherwise, `"main"` |
```

Comment:
```suggestion
| `capability:downscaleImage:workerType` | `"web worker"` if the browser supports Web Worker and offline canvas, otherwise, `"main"` |
```